### ISSUE TITLE
Allow migrate dev to reset if needed non-interactively

### ISSUE THREAD

**Issue by luxaritas** [prisma/prisma#19894]
## Problem
I want to enable a development environment where on app start, migrations are automatically applied. I would do this during a `docker compose up`. However, if drift is detected, it prompts the user. This prompt is liable to be hard to notice/respond to in the context of a larger `compose` setup, and since it's a development environment, data loss is considered acceptable and preferable over extra steps or confusion when trying to start or update an environment

## Suggested solution
Add an `--allow-reset` flag to allow the database to be reset if drift is encountered without prompting the user for confirmation

## Alternatives
* `db push` Push doesn't run data migrations, which is required in my use case - an extra `db seed` would be required which would mean the data would have to be duplicated in the seed script AND seeding would be forced to be idempotent as seeding would have to be run on every app start
* Don't automatically migrate

## Additional context
There appears to be a `--force` flag that would theoretically do this, but has been disabled. There is a link to a private issue, so I can't gain any more context on why it was removed: https://github.com/prisma/prisma/commit/28a069623aac2513c93f834734fdb05cd71f8293 FWIW, arguably `--allow-reset` (or `--accept-data-loss` to mirror db push) would be much clearer in intent and the destructive nature vs `--force`.

This appears to be a straightforward change, and I'd be happy to file a PR

**Comment by aswinramakrish** [comment]
I don't think this is a `kind/improvement`. To be honest, `migrate dev` is seriously broken! I have 2 database schemas and things were working fine as long as I wasn't using migrations. Once I started using `migrate dev`, migrations to one of the databases kept picking up changes (essentially dropping all the tables and re-creating them every single time) and wiping out the data completely. 

It's fine that Prisma doesn't support multi-database schema automatically, but at the very least, it could stop touching the data. Resetting databases and wiping out data makes it risky / almost unusable in a Production scenario. Again, to re-iterate.. this should be a priority (especially with all the confusion in the threads related to `--create-only`)

**Comment by luxaritas** [comment]
@aswinramakrish I think what you're describing is distinct from what this issue is asking for. The request here is the ability to reset without asking, which would make the situation you're describing worse.

**Comment by janpio** [comment]
Please open a separate issue or discussion @aswinramakrish and we can try to understand what is happening for your project. As @luxaritas explained, his use case is a very different one.

**Comment by paymog** [comment]
@janpio any updates on why force migrating was removed? The core logic for it still exists, but just the flag was disabled: https://github.com/prisma/prisma/blame/1eacb61c495b2e82ba8d8c77e2189c3df632a335/packages/migrate/src/commands/MigrateDev.ts#L136
